drivers/ethernet/eth_mcux.c

```diff
@@ -1276,116 +1264,227 @@ static void eth_mcux_error_isr(const struct device *dev)
+static void eth0_config_func(void)
 {
 #if DT_INST_IRQ_HAS_NAME(0, rx)
 	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, rx, irq),
 		    DT_INST_IRQ_BY_NAME(0, rx, priority),
-		    eth_mcux_rx_isr, DEVICE_GET(eth_mcux_0), 0);
+		    eth_mcux_rx_isr,
+		    DEVICE_GET(eth_mcux_0),
+		    0);
 	irq_enable(DT_INST_IRQ_BY_NAME(0, rx, irq));
 #endif
 
 #if DT_INST_IRQ_HAS_NAME(0, tx)
 	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, tx, irq),
 		    DT_INST_IRQ_BY_NAME(0, tx, priority),
-		    eth_mcux_tx_isr, DEVICE_GET(eth_mcux_0), 0);
+		    eth_mcux_tx_isr,
+		    DEVICE_GET(eth_mcux_0),
+		    0);
 	irq_enable(DT_INST_IRQ_BY_NAME(0, tx, irq));
 #endif
 
 #if DT_INST_IRQ_HAS_NAME(0, err_misc)
 	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, err_misc, irq),
 		    DT_INST_IRQ_BY_NAME(0, err_misc, priority),
-		    eth_mcux_error_isr, DEVICE_GET(eth_mcux_0), 0);
+		    eth_mcux_error_isr,
+		    DEVICE_GET(eth_mcux_0),
+		    0);
 	irq_enable(DT_INST_IRQ_BY_NAME(0, err_misc, irq));
 #endif
 
 #if DT_INST_IRQ_HAS_NAME(0, common)
 	IRQ_CONNECT(DT_INST_IRQ_BY_NAME(0, common, irq),
 		    DT_INST_IRQ_BY_NAME(0, common, priority),
-		    eth_mcux_dispacher_isr, DEVICE_GET(eth_mcux_0), 0);
+		    eth_mcux_dispacher_isr,
+		    DEVICE_GET(eth_mcux_0),
+		    0);
 	irq_enable(DT_INST_IRQ_BY_NAME(0, common, irq));
 #endif
 
 #if defined(CONFIG_PTP_CLOCK_MCUX)
+#if DT_NODE_HAS_STATUS(PTP_INST_NODEID(0), okay)
 	IRQ_CONNECT(DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, irq),
 		    DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, priority),
-		    eth_mcux_ptp_isr, DEVICE_GET(eth_mcux_0), 0);
+		    eth_mcux_ptp_isr,
+		    DEVICE_GET(eth_mcux_0),
+		    0);
 	irq_enable(DT_IRQ_BY_NAME(PTP_INST_NODEID(0), ieee1588_tmr, irq));
 #endif
+#endif
 }
+#endif /* DT_NODE_HAS_STATUS(DT_DRV_INST(0), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_DRV_INST(1), okay)
 
-#if defined(CONFIG_ETH_MCUX_1)
-static void eth_1_config_func(void);
+#if DT_INST_PROP(1, zephyr_random_mac_address) && \
+	NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(1))
+#error Conflict between 'local-mac-address' and 'zephyr,random-mac-address'
+#endif
 
-static struct eth_context eth_1_context = {
-	.base = ENET2,
+#if !NODE_HAS_VALID_MAC_ADDR(DT_DRV_INST(1))
+static void generate_eth1_mac(uint8_t *mac_addr)
+{
+#if DT_INST_PROP(1, zephyr_random_mac_address)
+	gen_random_mac(mac_addr,
+		       FREESCALE_OUI_B0, FREESCALE_OUI_B1, FREESCALE_OUI_B2);
+#else
+	/* Generate_unique_mac */
+	mac_addr[0] = FREESCALE_OUI_B0;
+	mac_addr[1] = FREESCALE_OUI_B1;
+	mac_addr[2] = FREESCALE_OUI_B2;
+	/* Trivially "hash" up to 128 bits of MCU unique identifier */
+#if defined(CONFIG_SOC_SERIES_IMX_RT)
+	uint32_t id = OCOTP->CFG1 ^ OCOTP->CFG2;
+#elif defined(CONFIG_SOC_SERIES_KINETIS_K6X)
+	uint32_t id = SIM->UIDH ^ SIM->UIDMH ^ SIM->UIDML ^ SIM->UIDL;
+#else
+#error "Unsupported SOC"
+#endif
+	mac_addr[0] |= 0x02; /* force LAA bit */
+
+	mac_addr[3] = id >> 8;
+	mac_addr[4] = id >> 16;
+	mac_addr[5] = id >> 0;
+
+	mac_addr[5] += 1;
```

Comment:
The idea was to have the exact same code and variabley for both enet and enet2 and then use the `DT_INST_FOREACH_STATUS_OKAY` macro. Unfortunately I was not able to get to something which compiles because of the many #if. If someone manage to do it we could get completely rid of the code duplication between the 2 instances.